pkg/reconciler/route/route_test.go

```diff
@@ -1073,20 +1067,30 @@ func TestUpdateDomainConfigMap(t *testing.T) {
 			fakerouteinformer.Get(ctx).Informer().GetIndexer().Add(route)
 			routeClient.Create(route)
 			if err := ctl.Reconciler.Reconcile(context.Background(), KeyOrDie(route)); err != nil {
-				t.Fatalf("Reconcile() = %v, want no error", err)
+				t.Fatal("Reconcile() =", err)
 			}
-			addResourcesToInformers(t, ctx, route)
+			addRouteToInformers(t, ctx, route)
 
-			if tc.apply != nil {
-				tc.apply(route, watcher)
-				fakerouteinformer.Get(ctx).Informer().GetIndexer().Add(route)
-				if _, err := routeClient.Update(route); err != nil {
-					t.Fatalf("Route.Update() = %v, want no error", err)
-				}
+			tc.apply(route, watcher)
+			fakerouteinformer.Get(ctx).Informer().GetIndexer().Add(route)
+			if _, err := routeClient.Update(route); err != nil {
+				t.Fatal("Route.Update() =", err)
 			}
 
 			if err := ctl.Reconciler.Reconcile(context.Background(), KeyOrDie(route)); err != nil {
-				t.Fatalf("Reconcile() = %v, want no error", err)
+				t.Fatal("Reconcile() =", err)
+			}
+
+			rl := fakerouteinformer.Get(ctx).Lister()
+			if err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
```

Comment:
Can we watch instead of polling or we don't care?

Reply:
Don't care it's test

Reply:
That's not what you said when I mentioned the overkill optimizations in some tests in another PR 😄

Reply:
It's not about optimization as much, since watch at some point is probably a loop like this :D
I am also not sure that informers export watch and watching client does not mean that the informer would capture it :)